File: backend/app/schemas/onboarding.py
from typing import Literal

from pydantic import BaseModel

OnboardingStep = Literal[
    "profile",
    "hauler_profile",
    "hauler_vehicle",
    "hauler_service_area",
    "hauler_documents",
    "hauler_verification",
    "done",
]


class OnboardingChecks(BaseModel):
    has_vehicle: bool
    has_service_area: bool
    has_insurance: bool
    has_drivers_license: bool
    has_background_check: bool


class OnboardingStatus(BaseModel):
    profile_complete: bool
    customer_ready: bool
    hauler_ready: bool
    next_step: OnboardingStep
    checks: OnboardingChecks

File: frontend/src/types/onboarding.ts
export type Role = "customer" | "hauler";

export type OnboardingStep =
  | "profile"
  | "hauler_profile"
  | "hauler_vehicle"
  | "hauler_service_area"
  | "hauler_documents"
  | "hauler_verification"
  | "done";

export interface OnboardingChecks {
  has_vehicle: boolean;
  has_service_area: boolean;
  has_insurance: boolean;
  has_drivers_license: boolean;
  has_background_check: boolean;
}

export interface OnboardingStatus {
  profile_complete: boolean;
  customer_ready: boolean;
  hauler_ready: boolean;
  next_step: OnboardingStep;
  checks: OnboardingChecks;
}
